### DISCUSSION TITLE
Shift + Scroll support plugin

### DISCUSSION THREAD

**Discussion by DjSt3rios** [idiotWu/smooth-scrollbar#367] [Show and tell]
Hello my friends, I made a little plugin that will allow users to scroll horizontally by holding the shift key. I hope someone finds this useful:

```ts
import { ScrollbarPlugin } from 'smooth-scrollbar';

export default class ShiftScrollPlugin extends ScrollbarPlugin {
  static pluginName = 'ShiftScroll';

  transformDelta(delta: { x: any; y: any; }, fromEvent) {
    return /wheel/.test(fromEvent.type) && fromEvent.shiftKey ? { x: delta.y, y: delta.x } : delta;
  }
}

```

You can then do something like this to activate it:

```js
Scrollbar.use(ShiftScrollPlugin, /* OverscrollPlugin, ... */);
Scrollbar.init(document.querySelector('#scroll'));
```

Big thanks to @idiotWu for showing me how to create this plugin 😁

If someone has trouble compiling and using the plugin, you can download the javascript from here and simply import it using the script tag: 
[Download](https://drive.google.com/file/d/12B62-sLSmwad196lOukJ6uz06ral1855/view?usp=sharing)

**Comment by dolphin-wood** [comment]
Thanks for sharing 🤗 !

Have you tried using `fromEvent.shiftKey` property directly to check if the shift key is pressed? Thought we might not need those event listeners defined in `onInit()` method.

See https://github.com/idiotWu/smooth-scrollbar/pull/366#issuecomment-877863319

**Reply by DjSt3rios** [reply]
I tried but there is no such property apparently. The error I get is:
`Property 'shiftKey' does not exist on type 'Event'`

**Reply by DjSt3rios** [reply]
Actually I just realized my IDE set the type to "Event" which is wrong, I tried removing the type and now it works. I will update it! Thanks 😁

**Comment by dolphin-wood** [comment]
Just created a demo for this plugin: https://codepen.io/idiotWu/pen/VwbborM
